keyberon/src/layout.rs

```diff
@@ -1638,6 +1645,13 @@ impl<'a, const C: usize, const R: usize, const L: usize, T: 'a + Copy + std::fmt
             .unwrap_or(self.default_layer)
     }
 
+    pub fn active_held_layers(&self) -> std::vec::Vec<usize> {
+        self.states
+            .iter()
+            .filter_map(State::get_layer)
+            .collect::<std::vec::Vec<_>>()
```

Comment:
Is it feasible to avoid allocating here by returning an iterator instead?

Reply:
sure

Reply:
[`ae512d3` (#752)](https://github.com/jtroo/kanata/pull/752/commits/ae512d30f1c9f71ba9294e2322eee99bb0bfc4a1)